invenio_rdm_records/resources/serializers/datacite/__init__.py

```diff
@@ -31,9 +31,10 @@ class DataCite43XMLSerializer(MarshmallowSerializer):
 
     def __init__(self, **options):
         """Constructor."""
+        encoder = options["encoder"] if "encoder" in options else schema43.tostring
```

Comment:
This might fail if options is passed as `None` (i.e. no `kwargs`):
```python
>>> options = None
>>> encoder = options["encoder"] if "encoder" in options else "test"
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: argument of type 'NoneType' is not iterable
```

In addition you can use `get` with a default value:

```suggestion
if options:
        encoder = options.get("encoder", schema43.tostring)
```